src/feature_extraction.py

```diff
@@ -477,6 +477,112 @@ def persist(self, filepath):
         joblib.dump(self.features, filepath)
 
 
+class PreviousApplicationFeatures(BaseTransformer):
+    def __init__(self, **kwargs):
+        self.features = None
+
+    @property
+    def feature_names(self):
+        feature_names = list(self.features.columns)
+        feature_names.remove('SK_ID_CURR')
+        return feature_names
+
+    def fit(self, X, prev_applications, **kwargs):
+        features = pd.DataFrame({'SK_ID_CURR': prev_applications['SK_ID_CURR'].unique()})
+
+        prev_applications_sorted = prev_applications.sort_values(['SK_ID_CURR', 'DAYS_DECISION'])
+        prev_applications_last_3 = prev_applications_sorted.groupby(by=['SK_ID_CURR']).tail(3)
```

Comment:
@pknut this should be parametrized so we could check which values work best (maybe 4,8,13 right?)

Reply:
@jakubczakon 100% agree, I am working on a better version of the code (parameterized and without repetitions).